nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
```

Comment:
Looks like transport node APIs are exposed via policy API `/infra/sites/{site-id}/enforcement-points/{enforcementpoint-id}/host-transport-nodes/`
Ideally we should leverage policy API. If we're sticking to MP API, host switch profiles should also be exposed via MP API